src/ros_filter.cpp

```diff
@@ -3528,6 +3728,207 @@ namespace RobotLocalization
+      }
+      sourceData_[topicName].bias_pose_ = measurement;
+      sourceData_[topicName].bias_valid_ = true;
+      sourceData_[topicName].last_bias_s_ = msg->header.stamp.toSec();
+      RF_VERBOSE("Bias data for topic " << topicName << " updated to " << measurement);
+    }
+
+    RF_VERBOSE("\n----- /RosFilter::biasPoseCallback (" << topicName << ") ------\n");
+  }
+
+  template<typename T>
+  void RosFilter<T>::biasTwistCallback(const geometry_msgs::TwistWithCovarianceStamped::ConstPtr &msg,
+                                   const CallbackData &callbackData,
+                                   const std::string &targetFrame)
+  {
+    const std::string &topicName = callbackData.topicName_;
+
+    RF_VERBOSE("------ RosFilter::biasTwistCallback (" << topicName << ") ------\n"
+             "Twist message:\n" << *msg);
+
+    Eigen::VectorXd measurement(STATE_SIZE);
+    Eigen::MatrixXd measurementCovariance(STATE_SIZE, STATE_SIZE);
+
+    measurement.setZero();
+    measurementCovariance.setZero();
+
+    // Make sure we're actually updating at least one of these variables
+    std::vector<int> updateVectorCorrected = callbackData.updateVector_;
+
+    // Prepare the twist data for inclusion in the filter
+    if (prepareTwist(msg, topicName, targetFrame, updateVectorCorrected, measurement, measurementCovariance))
+    {
+      // Save the bias data
+      if(sourceData_.find(topicName) == sourceData_.end())
+      {
+        RF_VERBOSE("Adding twist source data for topic " << topicName << "\n");
+        // Create the new data storage for this source
+        sourceData_.emplace(topicName, SourceData());
+      }
+      sourceData_[topicName].bias_twist_ = measurement;
+      sourceData_[topicName].bias_valid_ = true;
+      sourceData_[topicName].last_bias_s_ = msg->header.stamp.toSec();
+      RF_VERBOSE("Bias data for topic " << topicName << " updated to " << measurement);
+    }
+
+    RF_VERBOSE("\n----- /RosFilter::biasTwistCallback (" << topicName << ") ------\n");
+  }
+
+  template<typename T>
+  void RosFilter<T>::biasAccelerationCallback(const sensor_msgs::Imu::ConstPtr &msg, const CallbackData &callbackData,
+    const std::string &targetFrame)
+  {
+    const std::string &topicName = callbackData.topicName_;
+
+    RF_VERBOSE("------ RosFilter::biasAccelerationCallback (" << topicName << ") ------\n"
+             "Twist message:\n" << *msg);
+
+    Eigen::VectorXd measurement(STATE_SIZE);
+    Eigen::MatrixXd measurementCovariance(STATE_SIZE, STATE_SIZE);
+
+    measurement.setZero();
+    measurementCovariance.setZero();
+
+    // Make sure we're actually updating at least one of these variables
+    std::vector<int> updateVectorCorrected = callbackData.updateVector_;
+
+    // Prepare the twist data for inclusion in the filter
+    // Bias never includes the gravitational acceleration handling
+    if (prepareAcceleration(msg, topicName, targetFrame, false, updateVectorCorrected, measurement,
```

Comment:
I was wondering why not just use the `removeGravitationalAcc_` data. Makes more sense now.

Reply:
Yeah, leaving that in there actually causes -nans to occur, so I removed it. The expectation is that bias information is aligned with the sensor axes anyway (due to how it is provided to the localization system), so it would be incorrect for someone to provide gravity as a bias term anyway since that is in a world coordinate system, not the sensor coordinate system.